test/typescript/styles.spec.tsx

```diff
@@ -25,29 +26,25 @@ interface StyledComponentProps {
   text: string;
 }
 
-const Component: React.SFC<
-  StyledComponentProps & { classes: StyledComponentClassNames }
-> = ({ classes, text }) =>
-  <div className={classes.root}>
-    {text}
-  </div>;
-
-const StyledComponent = withStyles<
-  StyledComponentProps,
-  StyledComponentClassNames
->(styles)(Component);
+const StyledComponent = withStyles(styles)<StyledComponentProps>(
+  ({ classes, text }) => (
+    <div className={classes.root}>
+      {text}
+    </div>
+  )
+);
 
 <StyledComponent text="I am styled!" />;
 
 // Also works with a plain object
 
-const stylesAsPojo: StyleRules = {
+const stylesAsPojo: StyleRules<'root'> = {
```

Comment:
Does it mean people will have to duplicate the styles keys, one in the definition and one in the object?

Reply:
No, in fact you can now omit the type annotation completely here and get more type safety than you used to with the type annotation. You could also just annotate it as `StyleRules` as before and get the same (reduced) type safety that you had before (i.e., forgetting the particular style names and treating them as `string`s.)

Reply:
There should rarely be any reason for users to have to reference `StyleRules` explicitly in their code now.